app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -1,35 +1,60 @@
 package com.andrew.tasky.agenda.data.agenda
 
-import com.andrew.tasky.agenda.data.event.EventDatabase
-import com.andrew.tasky.agenda.data.reminder.ReminderDatabase
+import android.util.Log
+import com.andrew.tasky.agenda.data.database.AgendaDatabase
 import com.andrew.tasky.agenda.data.util.*
 import com.andrew.tasky.agenda.domain.AgendaRepository
+import com.andrew.tasky.agenda.domain.models.AgendaItem
 import com.andrew.tasky.auth.data.AuthResult
 import com.andrew.tasky.auth.util.getAuthResult
+import java.util.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 
 class AgendaRepositoryImpl(
     private val api: AgendaApi,
-    private val reminderDb: ReminderDatabase,
-    private val eventDb: EventDatabase
+    private val db: AgendaDatabase,
 ) : AgendaRepository {
 
-    override suspend fun getRemoteAgendaItems(
-        timezone: String,
-        time: Long
-    ): AuthResult<Unit> {
-        return getAuthResult {
-            val response = api.getAgendaItems(timezone = timezone, time = time)
-            val tasks = response.taskDtos.map { task ->
-                task.toTask()
+    override fun getAgendaItems(): Flow<List<AgendaItem>> {
+        val localReminders = db.getReminderDao().getReminders().map {
+            it.map { reminderEntity ->
+                reminderEntity.toReminder()
             }
-            response.reminderEntities.map { reminder ->
-                reminderDb.getReminderDao().upsert(reminder)
-            }
-            // TODO Make event and task db connect to api
         }
+        return localReminders
+        // Todo return task and event agenda items
+    }
```

Comment:
If all you do is create a variable and then return it, I'd just return it right away, since there's no gain in having it. Variables make sense if you need them multiple times or they make something more readable, but here the function name kinda tells you what it represents :)